### DISCUSSION TITLE
Bring back Andy! or some other logo for C#

### DISCUSSION THREAD

**Discussion by HappyTreesDev** [dotnet/csharplang#2708] [General]
A long time ago C# had a logo called Andy based off of the creator. It would be great to bring back the logo or make a new one to represent the language. The official logo would help with branding of the language and its usage all over the place. Maybe a logo that adds space to support the compiler used or the framework used along with the language.

**Comment by Gnbrkm41** [comment]
Whatever the new logo might be, it must be better than the [Gopher](https://www.bing.com/search?q=Golang+gopher) :^)

**Reply by littletrainee** [reply]
Gopher is adorable. I wish C# had something similar. Maybe GitHub's mascot could be a reference point.

**Comment by orthoxerox** [comment]
Wasn't there a Superman/gemstone-styled logo already?

**Comment by LokiMidgard** [comment]
Oreyl uses the Grey Crowned Crane (African Crowned Crane)

**Comment by donaldnevermore** [comment]
Or just a simple "C#" logo like this?

![csharp-dotnet](https://user-images.githubusercontent.com/29004603/75462714-9dd79b80-59bf-11ea-8e6b-575765733340.png)

It's great to have an official logo even the simplest one.

**Reply by smoothdeveloper** [reply]
F# foundation made the logo based on `|>` operator (and an official poll IIRC).

What shape / piece of syntax would convey C#ness along the same line?

For example, the fat arrow `=> {};` is for me, what defined C# when it made inches toward being more expressive, `var` was also for me important point in time to start gaining appreciation for type inference. IMO those are still core to the idioms of today's C#.

Maybe the community can pool over what defines the most C# in similar fashion it came for F# community?

Another approach is to convey the essence of language, like Rust with the rusty gear, by finding real life object that sums C# up.

I think using the language designer as mascot is not a good choice, languages transcends whoever summoned it (no offense to Andy, we love you!).

**Reply by mattwar** [reply]
C# is best represented by { get; }

**Reply by orthoxerox** [reply]
Or `required Foo { get; init; }` these days. Not exactly the syntax the language should be proud of.

**Comment by amal-stack** [comment]
Something cute and plushy like Dart's and Flutter's [Dash](https://docs.flutter.dev/dash) would be nice :)

![Dash](https://docs.flutter.dev/assets/images/dash/BigDashAndLittleDash.png)

<sup>From https://docs.flutter.dev/assets/images/dash/BigDashAndLittleDash.png</sup>

**Comment by HaloFour** [comment]
I vote for this:

![image](https://github.com/dotnet/csharplang/assets/10102290/80ab7f7f-3894-451e-87a0-4e27e45d0c43)

/cc @CyrusNajmabadi

**Comment by littletrainee** [comment]
Since official want to make it open-source, is it possible to extract some features for the public to vote on whether to adopt them, including mascots and direct binary files instead of using .NET?

**Reply by HaloFour** [reply]
I believe that would be those features that the C# team put forward as open for community contributions.  Otherwise, the feedback through this repo (and many other venues) are the method through which the public can "vote" for features they would like to see.  Ultimately it's up to the team whether or not anything is adopted, and any proposal requires team members to "champion" any language features through the design process.  The language specification, with which this repo is concerned, doesn't really care about mascots or to what format the compiler emits, though.

**Comment by leo-costa** [comment]
I vote for this :
![image](https://github.com/dotnet/csharplang/assets/32940986/323f51f8-dbca-4dc9-8e9f-308c5490a2a1)

**Reply by CeliaCasanovas** [reply]
i'm seconding this. C# is for C#apybara!

**Reply by LokiMidgard** [reply]
While this could confused with [capybara-language](https://github.com/capybara-language/compiler) or the [capybara test automatisation software](https://en.wikipedia.org/wiki/Capybara_(software)). I'm totally for that :) 


![CapybaraOrangeGIF](https://github.com/dotnet/csharplang/assets/389101/202a6993-3310-4031-a421-cc23031ac1a6)

### C-Sharpybara!

**Comment by Jure-BB** [comment]
Because they C# 👀 

![image](https://github.com/dotnet/csharplang/assets/6940357/a87b2b4e-9328-4a93-b355-9572588396ec)

**Reply by littletrainee** [reply]
This is great! I can imagine using an eagle as a mascot, it would definitely add a lot of fun. If we include it in the documentation or API, it will certainly reduce the seriousness and make it more enjoyable for users.

**Reply by Zodt** [reply]
Sorry, but the first association from this eagle toy is like this)))
![image](https://github.com/dotnet/csharplang/assets/9018541/098f0118-8d1c-42b9-bde2-412015eb790d)

**Comment by perisicnikola37** [comment]
Well, these days it's common to use artificial intelligence to generate this kind of staff. Not so bad, don't you think?
<img src="https://github.com/user-attachments/assets/43280758-6da3-4ac7-aa1c-1ce0f975acbc" height="400" width="auto" />

**Comment by elibroftw** [comment]
Chameleon; Mechanical or just full purple
<img width="878" height="862" alt="552683f5-042f-4357-871c-b203b492dabe Background Removed" src="https://github.com/user-attachments/assets/e69ffee8-fe83-4aa8-9d86-cfa93920e1da" />

**Reply by Zodt** [reply]
A nice detail: the Terminator literally says “I’ll be back” — and C# with source generators seems to say “I’ll generate myself back” 😄
